addons/payment_paymob/models/payment_provider.py

```diff
@@ -0,0 +1,314 @@
+        groups='base.group_system',
+    )
+    paymob_hmac_key = fields.Char(string="Paymob HMAC Key", required_if_provider='paymob')
+    paymob_api_key = fields.Char(string="Paymob API Key", required_if_provider='paymob')
+    paymob_access_token = fields.Char(
+        string="Paymob Access Token",
+        help="The short-lived token used to access Paymob APIs",
+        groups='base.group_system',
+    )
+    paymob_access_token_expiry = fields.Datetime(
+        string="Paymob Access Token Expiry",
+        help="The moment at which the access token becomes invalid.",
+        default='1970-01-01',
+        groups='base.group_system',
+    )
+
+    # ==== CONSTRAINT METHODS === #
+
+    @api.constrains('available_currency_ids', 'state')
+    def _limit_available_country_currency_ids(self):
+        for provider in self.filtered(lambda p: p.code == 'paymob'):
+            if len(provider.available_currency_ids) > 1 and provider.state != 'disabled':
+                raise ValidationError(_("Only one currency can be selected by Paymob account."))
+
+    # === COMPUTE METHODS === #
+
+    @api.depends('available_currency_ids')
+    def _compute_paymob_account_country(self):
+        for provider in self.filtered(lambda p: p.code == 'paymob'):
+            if len(provider.available_currency_ids) == 1:
+                provider.paymob_account_country_id = self.env['res.country'].search(
+                    [('code', '=', provider._get_country_from_currency())], limit=1)
+
+    @api.onchange('paymob_account_country_id')
+    def _onchange_paymob_account_country(self):
+        for provider in self.filtered(lambda p: p.code == 'paymob'):
+            provider.available_currency_ids = self.env['res.currency'].with_context(
+                active_test=False,
+            ).search([('name', '=', provider._get_paymob_account_currency())], limit=1)
+
+    # === ACTION METHODS === #
+
+    def action_sync_paymob_payment_methods(self):
+        """ Synchronize the payment methods with the ones on the paymob portal, the integration_name
+        needs to be set to be able to communicate with the `payment_method.code` when the intention
+        is created.
+
+        :return: None
+        """
+        base_url = self.get_base_url()
+        redirect_url = urls.url_join(base_url, PaymobController._return_url)
+        webhook_url = urls.url_join(base_url, PaymobController._webhook_url)
+        endpoint = '/api/ecommerce/integrations'
+        is_live = self.state == 'enabled'
+        params = {
+            'is_plugin': 'true',
+            'page_size': 500,
+            'is_deprecated': 'false',
+            'is_standalone': 'false',
+            'is_live': json.dumps(is_live),
+        }
+        self.paymob_access_token = None
+        paymob_payment_methods = self._paymob_make_request(
+            endpoint,
+            params=params,
+            method='GET',
+            is_client_request=False,
+        )['results']
```

Comment:
the endpoint is used only here, no need a variable for that
```suggestion
        paymob_payment_methods = self._paymob_make_request(
            '/api/ecommerce/integrations',
            params=params,
            method='GET',
            is_client_request=False,
        )['results']
```